### PR TITLE
Create escalated PeerReview when all reviews are in with no consensus

### PR DESCRIPTION
**Pull request by aoby**

### CODE REVIEW

File: dashboard/app/models/peer_review.rb [hunk 1]
```diff
@@ -126,15 +126,17 @@ def mark_user_level
...
       user_level.update!(best_result: Activity::REVIEW_ACCEPTED_RESULT)
       update_column :audit_trail, append_audit_trail("COMPLETED by user id #{reviewer_id}")
     elsif reviews.all?(&:rejected?)
       user_level.update!(best_result: Activity::REVIEW_REJECTED_RESULT)
       update_column :audit_trail, append_audit_trail("REJECTED by user id #{reviewer_id}")
     else
+      # No consensus: escalate the review (i.e. create an escalated review based on this one)
+      escalated_review = dup
+      escalated_review.assign_attributes(status: 'escalated', reviewer: nil)
+      escalated_review.save!
```
Comment (aoby) [inline dashboard/app/models/peer_review.rb:140]: @mehalshah when this new escalated peer review is reviewed, will this callback again log no consensus and re-escalate? Should the above logic be different? Or are `escalated` reviews handled differently somewhere?

Maybe we need more unit tests here, for the next step, reviewing (i.e. updating status to accepted or rejected) the escalated review?

File: dashboard/test/models/peer_review_test.rb [hunk 2]
```diff
@@ -367,13 +367,18 @@ class PeerReviewTest < ActiveSupport::TestCase
     assert reviews.last.audit_trail.lines.last.include? "REJECTED by user id #{users.last.id}"
   end
 
-  test 'no consensus reviews are logged to the audit trail' do
+  test 'no consensus reviews log to the audit trail and escalate' do
     track_progress @level_source.id
     reviews = PeerReview.last(2)
     users = create_list :user, 2
 
     reviews[0].update!(reviewer: users[0], status: 'accepted')
```
Comment (mehalshah) [inline dashboard/test/models/peer_review_test.rb]: I'd suggest having an assert_no_creates around review[0]

File: dashboard/test/models/peer_review_test.rb [hunk 3]
```diff
@@ -367,13 +367,18 @@ class PeerReviewTest < ActiveSupport::TestCase
...
-    reviews[1].update!(reviewer: users[1], status: 'rejected')
+    assert_creates PeerReview do
+      reviews[1].update!(reviewer: users[1], status: 'rejected')
+    end
+    escalated_review = PeerReview.last
+    assert_nil escalated_review.reviewer
+    assert_equal 'escalated', escalated_review.status
```
Comment (mehalshah) [inline dashboard/test/models/peer_review_test.rb:383]: Additionally we should assert that submitter_id is set, from_instrctor is true, script, level, level_source are set, and data is nil

File: dashboard/app/models/peer_review.rb (hunk 1 above)
Comment (mehalshah) [reply dashboard/app/models/peer_review.rb:140]: Line 102 handles this.

While you are here, could you append to the audit trail in the line 102 block?

**Comment by aoby** [comment]
Updated, PTAL

File: dashboard/test/models/peer_review_test.rb [hunk 4]
```diff
@@ -372,19 +372,39 @@ class PeerReviewTest < ActiveSupport::TestCase
...
+    assert_equal @level_source, escalated_review.level_source
+    assert_nil escalated_review.data
 
     # 1 line for the assignment, 1 for the review, 1 for the no consensus
     assert_equal 3, reviews.last.audit_trail.lines.count
     assert reviews.last.audit_trail.lines.last.include? "NO CONSENSUS after review by user id #{users.last.id}"
   end
 
+  test 'instructor reviews log to the audit trail' do
+    instructor = create :teacher
```
Comment (mehalshah) [inline dashboard/test/models/peer_review_test.rb]: Nit - instead of :teacher, can you create a :plc_reviewer?

**Pull request merged**